subsystems/ElevatorBase.java

```diff
@@ -191,20 +191,32 @@ public void stop() {
     @Override
     public void useOutput(double output, TrapezoidProfile.State setpoint) {
         double ff = ffController.calculate(setpoint.position, setpoint.velocity);
-        SmartDashboard.putNumber("FEEDFORWARD", ff);
-        SmartDashboard.putNumber("OUTPUT USED", output);
-        SmartDashboard.putNumber("SETPOINT USED", setpoint.position);
-        SmartDashboard.putNumber("GOAL USED", pidController.getGoal().position);
-        SmartDashboard.putBoolean("AT SETPOINT", pidController.atGoal());
-
-        if(SmartDashboard.getBoolean("OUTPUT ON?", true)) {
-            setVoltage(output + ff);
-        }
-        else if(!SmartDashboard.getBoolean("OUTPUT ON?", true) && getMeasurement() > constants.softLimits[1] && output > 0) {
-            setVoltage(0);
+        //TODO: remove debug prints once we know this code works
+        SmartDashboard.putNumber("ELEVATOR FEEDFORWARD", ff);
+        SmartDashboard.putNumber("ELEVATOR OUTPUT USED", output);
+        SmartDashboard.putNumber("ELEVATOR SETPOINT USED", setpoint.position);
+        SmartDashboard.putNumber("ELEVATOR GOAL USED", pidController.getGoal().position);
+        SmartDashboard.putBoolean("ELEVATOR AT SETPOINT", pidController.atGoal());
+
+        /** if the driver has set output on, useOutput. */
+        if(SmartDashboard.getBoolean("ELEVATOR OUTPUT ON?", true)) {
+            /** output should be feedforward + calculated PID. */
+            /** if the limit switch is pressed and the elevator is powered to move downward, set the voltage to 0 */
+            if(getLimitSwitchValue(constants.switchPorts[0]) && output < 0) {
+                setVoltage(0);
```

Comment:
We don't necessarily wanna turn off the elevator when the limit switch is triggered.